src/test/ui/span/lint-unused-unsafe.rs

```diff
@@ -1,7 +1,11 @@
 // Exercise the unused_unsafe attribute in some positive and negative cases
 
-// revisions: mir thir
-// [thir]compile-flags: -Zthir-unsafeck
+
+// edition:2018
+// revisions: mir
+
+// // revisions: mir thir
+// // [thir]compile-flags: -Zthir-unsafeck
```

Comment:
I suspect this may have been in place to verify some sort of possibly breaking change that has occurred between 2015 and 2018 editions. Are we sure the `thir` mode can be removed at this point?

Either way, lets not have commented-out code, or if we absolutely must have any, add a comment why it is here and what is blocking it from being uncommented.

Reply:
AFAIK, `-Zthir-unsafeck` is a *new* version of the unsafety check that’s in development (though at the moment mostly inactive). I do actually have managed to integrate the new version of the `unused_unsafe` check from this PR into the thir-version, too, as mentioned in https://github.com/rust-lang/rust/pull/93678#issuecomment-1031743297, which re-activates this test case. But IMO it makes more sense to focus on the stable / non-experimental mir-version. I mean if you _want_ to review both together, I might as well add that into this PR as well, even then, we should probably finish discussing the behavior first before I’ll update both implementations accordingly.

Reply:
Of course, if we keep this for a separate PR, I can (and probably should) add an explanatory comment.

Reply:
Yeah, I don't think its strictly necessary for us to make sure the two implementations are in line in this same PR, nor do I think the improvements to a stable implementation should be blocked by an unstable one. I would rather not review gigantic PRs either. That said, given your explanation, I feel it is important the other implementation receives coverage still.

Maybe the most straightforward way to do so would be to split the test into two files for the time being?

Reply:
Right, splitting the test for now might be a reasonable option, in order to better document that the `thir` check is unchanged so far (and in order to not make test coverage worse than before). I think I can do that tomorrow, I'll add a comment noting that it's the goal to "reunify" the tests later.

Reply:
I’ve split them now